frontend/model/contracts/group.js

```diff
@@ -302,7 +297,11 @@ sbp('chelonia/defineContract', {
       const pendingMembers = {}
       for (const inviteId in invites) {
         const invite = invites[inviteId]
-        if (invite.status === INVITE_STATUS.VALID && invite.creator !== INVITE_INITIAL_CREATOR) {
+        if (
+          invite.status === INVITE_STATUS.VALID &&
+          invite.creator !== INVITE_INITIAL_CREATOR &&
+          invite.expires >= Date.now()
```

Comment:
Nowhere inside the contract code — except for sideEffects — can you call `Date.now()`.

Getters and process functions must always behave the same no matter where they are run. `Date.now()` is unique to each individual client that calls it. Must be removed here.

Reply:
Understand. Nice catch...

Reply:
But if we think that way, what about the `createdDate` of metadata.

```
createdDate: new Date().toISOString()
```

This depends on the creator's local date and if it is not correct, the wrong date will be added to contract.
How can we handle this?

Reply:
I think this is important problem... We should use the same datetime of backend, shouldn't we?

Reply:
> This depends on the creator's local date and if it is not correct, the wrong date will be added to contract.
How can we handle this?

Issue #531 is actually about this. A `/time` route exists in the backend. The only issue is that we currently don't use it. We would have to change the contract `create()` function for metadata to be asynchronous for this to work...

(Although it should be noted that we cannot guarantee that a hacked client won't still send the wrong date.)

However, all of this is besides the point here - we cannot use `Date.now()` here regardless.

Inside of the contract getters and contract process functions you can only use the `data` and `meta`data.

Reply:
@taoeffect, I can't find the better place and solution to check expiry.
Sending all pending users (no matter their invites are expired) to the frontend and checking their expiry in frontend components, doesn't seems to be a good practice.
I can also find the `isNewMember` function in `groupMembersSorted` getter which using Date.now().
Line 436 in frontend/model/state.js

The best solution is to create a sbp function to get the Date.now() on backend and use it. Of course, I am not sure this could be also acceptable because we are not using `sbp` functions inside process and getters.

If it is the solution of the #531, and also if we should do that way, I think I can skip it in this PR and resolve it creating a new PR from the #531.

Reply:
> The best solution is to create a sbp function to get the Date.now() on backend and use it.

... this really isn't necessary.

If all you want is to filter the pending members, you can modify this getter to include the expiry, and create a new getter in state.js that filters the results of this getter. The getters in state.js are allowed to use `Date.now()` and anything else.

Reply:
Okay. Understood why the solution should be like it.